bin/oneoff/backfill_data/user_cap_state_columns

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env ruby
+
+# This script migrates CAP (Child Account Policy) data from properties to separate indexed columns.
+
+require 'ruby-progressbar'
+require_relative '../../../dashboard/config/environment'
+
+# @see https://github.com/code-dot-org/code-dot-org/pull/52244
+CAP_RELEASE_DATE = DateTime.parse('2023-06-09')
+
+users = User.with_deleted.where(current_sign_in_at: CAP_RELEASE_DATE..)
+total_updated = 0
+failed_user_ids = []
+
+update_progress_title = proc {|num_updates = 1| "Processed[%c/%C]: |%W| Updated: #{total_updated += num_updates} %a"}
+progress_bar = ProgressBar.create(total: users.count, format: update_progress_title.call(0))
+
+users.select(
+  :id,
+  :cap_state,
+  :cap_state_date,
+  "properties->>'$.child_account_compliance_state' AS original_cap_state",
+  "properties->>'$.child_account_compliance_state_last_updated' AS original_cap_state_date"
+).find_each(batch_size: 10_000) do |user|
+  next if user.cap_state || user.original_cap_state.blank?
+
+  user.update_columns(
```

Comment:
I believe we do want to make sure each of these updates happens in an individual transaction; does `update_columns` do that for us, or do we need to explicitly wrap it in an `ActiveRecord::Base.transaction` block?

Reply:
Correct. Since we use InnoDB with [autocommit](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_autocommit) mode enabled (`SELECT @@autocommit;`), each SQL statement forms a single transaction on its own. See: https://dev.mysql.com/doc/refman/8.0/en/innodb-autocommit-commit-rollback.html#:~:text=In%20InnoDB,%20all%20user%20activity%20occurs%20inside%20a%20transaction

Reply:
Great, thanks!